### ISSUE TITLE
support/historyarchive: Fix idle timeout when streaming buckets

### ISSUE THREAD

**Issue by bartekn** [stellar/go#1649]
`XdrStream.ReadOne` method returns this error from time to time:
```
stream error: stream ID 311; PROTOCOL_ERROR
```
This is happening when there's a delay processing entries from a bucket. There can be multiple reasons for delays: preloaded entries are being processed, a batch of processed entries is being committed to a DB, etc. #1648 added a temporary fix for this issue (decreasing `preloadedEntries` to 20000 to make the delay between streaming batches shorter) but we need to find a permanent fix. Possible solutions:
* Send heartbeat/ping to a server when there are delays in streaming (if possible).
* When connection is closed, reconnect and rewind (if possible).
* When preloading entries start creating a new batch when the existing batch is being consumed. Not sure if it's a good idea as it requires to use a package in a specific way.
* Increase idle timeout in the history archive server (however, this may affect users of non-SDF archives).
* This can help: #1613 but fixes this issue in Horizon only.

**Comment by ire-and-curses** [comment]
After implementing #1613 are we still experiencing this in Horizon? I can see that adding some kind of retry logic could make this robust. I'm wondering how urgent it is.

**Comment by bartekn** [comment]
I think this is fixed in Horizon after #1613 as state ingestion takes a few minutes. However, would be great to fix it so users of `exp/ingest` don't experience it. Probably lower priority but good to have before `exp/ingest` beta is released.

**Comment by tamirms** [comment]
fixed in https://github.com/stellar/go/pull/1899
